enhancements/sig-architecture/63-hosted-addon/README.md

```diff
@@ -0,0 +1,262 @@
+       |       |  | deployment:         |  | deployment:         |          |       |                          
+  apply|       |  |  registration-agent |  |  hello-addon-agent  |          |       |                          
+       |       |  |  work-agent         |  |                     <-------------------                          
+       |       |  |                     |  |                     |          |                                  
+       |       |  +---------------------+  +---------------------+          |                                  
+       |       |                                      |connect              |                                  
+       |       +--------------------------------------|---------------------+                                  
+       |                                              |                                                        
+       |       +--------------------------------------v---------------------+                                  
+       |       | managed-cluster                                            |                                  
+       |       |                                                            |                                  
+       |       | +---------------------+   +---------------------+          |                                  
+       |       | |CRDs:                |   |Configurations:      |          |                                  
+       |       | | clusterClaim        |   | role,rolebinding,sa |  etc.    |                                  
+       |       | | appliedmanifestwork |   +----------^----------+          |                                  
+       |       | +---------------------+              |                     |                                  
+       |       +--------------------------------------|---------------------+                                  
+       |                                              |                                                        
+       |                                              |                                                        
+       -----------------------------------------------|                                                        
+```
+
+### Guidance for transforming Default mode addon to hosted mode
+
+Here are some guidelines for transforming addons, which are built with addon-framework in Default mode, to Hosted mode. 
+After transforming, the addon should be deployed successfully either in Default mode or Hosted mode.
+
+1. Consider carefully whether the addon can run in Hosted mode. Will addon functions be affected when addon agent is 
+running outside the managed cluster? If the answer is no, stop.
+2. Refactor the code of the addon agent to accept a flag like `managed-kubeconfig`, and use the kubeconfig received 
+with the flag to build the managed cluster kube client, and use this client to process resources on the managed cluster.
```

Comment:
this should be different from the `local-kubeconfig` or `in-cluster kubeconfig` which is mainly used for leader election, and lease update. right?

Reply:
Great point! If I remember right, the addon-framework usually uses a lease for addon health on the managed cluster. In Hosted mode, I think that would be on the hosting cluster? This is another place where the footprint on the managed cluster is reduced 👍 

But also, there is a backup way to put the lease on the hub cluster, in case the managed cluster doesn't support that API (eg openshift 3.11). Would that need to be considered here as well, for a potential 3 kubeconfigs? Or can we require the hosting cluster to support that API?

Reply:
yes, Great point. I think the lease should be on the hosting cluster.